src/poker/EquityEvaluator.java

```diff
@@ -40,52 +30,65 @@ public HashMap<String, String> loadHashMapFromFile(String filePath) {
+//        HashMap<String, String> instaResult = new HashMap<>();
+//        for (int i = 0; i < 8; i++) {
+//            for (int j = i + 1; j < 8; j++) {
+//                for (int k = j + 1; k < 8; k++) {
+//                    for (int l = k + 1; l < 8   ; l++){
+//                        List<Card> player1Hand = new ArrayList<>();
+//                        List<Card> player2Hand = new ArrayList<>();
+//                        player1Hand.add(deck.cardList.get(i));
+//                        player1Hand.add(deck.cardList.get(j));
+//                        player2Hand.add(deck.cardList.get(k));
+//                        player2Hand.add(deck.cardList.get(l));
+//                        player1Hand.sort(Comparator.comparing(Card::toString));
+//                        player2Hand.sort(Comparator.comparing(Card::toString));
+//                        List<Card> bothPlayers = new ArrayList<>(player1Hand);
+//                        bothPlayers.addAll(player2Hand);
+//                        List<Card> bothPlayers2 = new ArrayList<>(player2Hand);
+//                        bothPlayers2.addAll(player1Hand);
+//                        String equity = calculateEquityPreFlop(player1Hand, player2Hand, new Deck()).toString();
+//                        instaResult.put(bothPlayers.toString(), equity);
+//                        instaResult.put(bothPlayers2.toString(),equity);
+//                    }
+//                }
+//            }
+//        }
+//        saveHashMapToFile(instaResult, "equity_data.ser");
+//        return instaResult;
+//    }
+    public List<String> createPossbileKeys(List<Card> h1, List<Card>h2){
         h1.sort(Comparator.comparing(Card::toString));
         h2.sort(Comparator.comparing(Card::toString));
-        List<Card> list = new ArrayList<>(h1);
-        list.addAll(h2);
-
-        System.out.println(equityMap.get(list.toString()));
+        List<Card> bothPlayers = new ArrayList<>(h1);
+        bothPlayers.addAll(h2);
+        List<Card> bothPlayers2 = new ArrayList<>(h2);
+        bothPlayers2.addAll(h1);
+        List<String> equitiesPlayers = new ArrayList<>();
+        equitiesPlayers.add(bothPlayers.toString());
+        equitiesPlayers.add(bothPlayers2.toString());
+        return equitiesPlayers;
+    }
+    private List<Double> parseEquity(String equityString) {
+        equityString = equityString.replace("[", "").replace("]", "");
+        String[] equities = equityString.split(",");
+        List<Double> equityList = new ArrayList<>();
+        for (int i = 0; i < equities.length; i++) {
+            equityList.add(Double.parseDouble(equities[i].trim()));
+        }
+        return equityList;
     }
 
-    public List<Double> calculateEquityTest(List<Card> hand1, List<Card> hand2) {
-
+    public List<Double> calculateEquityPreFlop(List<Card> hand1, List<Card> hand2, Deck deck){
+        Map<String, String> equityPlayers = loadHashMapFromFile("equity_data.ser");
```

Comment:
takie ladowanie nie powinno byc w funkcji, bo wczytywanie plikow trwa dluzej niz petle itp, a po drugie bedzie sie wczywalo za kazdym razem kiedy wejdziemy w ta funckje. Idealne miejsce na to jest w konstrutkorze bo sie wczyta raz o pozniej mozna z tego korzystac